src/App.js

```diff
@@ -24,62 +24,87 @@ import getToken from './lib/tokens';
 
 function App() {
   const [listId, setListId] = useState(null);
-  const [userToken, setUserToken] = useState('');
 
-  function saveToken(e) {
-    e.preventDefault();
+  function createList() {
     const token = getToken();
-    localStorage.setItem('token', token);
-    setUserToken(token);
+
+    return db
+      .collection('lists')
+      .add({ token: token })
+      .then((results) => {
+        const { id } = results;
+        if (id === undefined) throw new Error('Failed to create list.');
+        return id;
+      })
+      .then((newListId) => {
+        // list successfully created in db, so update listId state and save token to localStorage
+        setListId(newListId);
+        localStorage.setItem('token', token);
+        return true;
+      });
```

Comment:
I think it would be good practice to add a check in here, to make sure the token doesn't already exist in the database before adding it as a unique identifier for an account. I think we could do it pretty easily with our new isTokenValid function.
```suggestion
    return isTokenValid(token)
      .then((listExists) => {
        if(listExists) {
          createList(); // this token is taken, so try again
        } else {
          return db
          .collection('lists')
          .add({ token: token })
          .then((results) => {
            const { id } = results;
            if (id === undefined) throw new Error('Failed to create list.');
            return id;
          })
          .then((newListId) => {
            // list successfully created in db, so update listId state and save token to localStorage
            setListId(newListId);
            localStorage.setItem('token', token);
            return true;
          });
        }
      });
```